internal/service/encryptionatrestprivateendpoint/resource.go

```diff
@@ -0,0 +1,110 @@
+//nolint:gocritic
```

Comment:
@AgustinBettati lmk if you have any thoughts on this approach, was thinking if it may be possible to ensure this is removed before merge to master, I have added that as an item for tech discussion meeting next week